### PR TITLE
Fix compiler warnings.

### PR DESCRIPTION
**Pull request by jaykrell**
intrinsics.c: In function 'mini_emit_inst_for_method':
    intrinsics.c:1580:11: warning: 'result' may be used uninitialized in this function [-Wmaybe-uninitialized]
         *dest = result;
         ~~~~~~^~~~~~~~


    debugger-engine.c: In function 'set_bp_in_method':
    debugger-engine.c:354:11: warning: variable 'code' set but not used [-Wunused-but-set-variable]
      gpointer code;
               ^~~~


    tramp-arm64-gsharedvt.c: In function 'mono_arch_get_gsharedvt_trampoline':
    tramp-arm64-gsharedvt.c:217:54: warning: variable 'callee_stack_area_offset' set but not used [-Wunused-but-set-variable]
      int caller_reg_area_offset, callee_reg_area_offset, callee_stack_area_offset;
                                                          ^~~~~~~~~~~~~~~~~~~~~~~~


    mini-arm64-gsharedvt.c: In function 'mono_arch_get_gsharedvt_call_info':
    mini-arm64-gsharedvt.c:142:29: warning: variable 'gsig' set but not used [-Wunused-but-set-variable]
      MonoMethodSignature *sig, *gsig;
                                 ^~~~


    interp/mintops.c: In function 'mono_interp_dis_mintop':
    interp/mintops.c:87:37: warning: format '%lld' expects argument of type 'long long int', but argument 3 has type 'long int' [-Wformat=]
       g_string_append_printf (str, " %lld", (gint64)READ64 (ip + 1));


    In file included from object.c:40:0:
    object.c: In function 'mono_class_proxy_vtable':
    ../../mono/metadata/profiler-private.h:191:4: warning: 'pvt' may be used uninitialized in this function [-Wmaybe-uninitialized]
        mono_profiler_raise_ ## name args; \
        ^~~~~~~~~~~~~~~~~~~~
    object.c:2338:19: note: 'pvt' was declared here
      MonoVTable *vt, *pvt;
                       ^~~


    w32file-unix.c: In function 'mono_w32file_get_file_size':
    w32file-unix.c:4994:19: warning: 'length_hi' may be used uninitialized in this function [-Wmaybe-uninitialized]
      return length | ((gint64)length_hi << 32);
                       ^~~~~~~~~~~~~~~~~


    interp/interp.c: In function 'interp_entry.constprop':
    interp/interp.c:1847:3: warning: 'orig_domain' may be used uninitialized in this function [-Wmaybe-uninitialized]
       mono_threads_detach_coop (orig_domain, &attach_cookie);
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


    image.c:1351:20: warning: variable 'header' set but not used [-Wunused-but-set-variable]
      MonoDotNetHeader *header;
                        ^~~~~~


    exception.c:747:1: warning: 'mono_new_exception_argument_out_of_range' defined but not used [-Wunused-function]
     mono_new_exception_argument_out_of_range (const char *arg, MonoError *error)
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


    os-event-unix.c: In function 'mono_os_event_wait_multiple':
    os-event-unix.c:195:12: warning: 'start' may be used uninitialized in this function [-Wmaybe-uninitialized]
        elapsed = mono_msec_ticks () - start;
        ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~


    In file included from ../../mono/eglib/glib.h:45:0,
                     from os-event.h:9,
                     from os-event-unix.c:11:
    ../../mono/eglib/eglib-remap.h:146:25: warning: 'data' may be used uninitialized in this function [-Wmaybe-uninitialized]
     #define g_ptr_array_add monoeg_g_ptr_array_add
                             ^~~~~~~~~~~~~~~~~~~~~~
    os-event-unix.c:121:19: note: 'data' was declared here
      OSEventWaitData *data;
                       ^~~~

    marshal.c: In function 'mono_marshal_get_runtime_invoke_full':
    marshal.c:2662:8: warning: 'sig_cache' may be used uninitialized in this function [-Wmaybe-uninitialized]
        res = (MonoMethod *)g_hash_table_lookup (sig_cache, sig_key);
        ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


    sgen-mono.c:260:1: warning: 'install_noilgen' defined but not used [-Wunused-function]
     install_noilgen (void)


    In file included from sgen-new-bridge.c:31:0:
    sgen-dynarray.h:331:1: warning: 'dyn_array_ptr_set_all' defined but not used [-Wunused-function]
     dyn_array_ptr_set_all (DynPtrArray *dst, DynPtrArray *src)
     ^~~~~~~~~~~~~~~~~~~~~


    sgen-dynarray.h:248:1: warning: 'dyn_array_ptr_set' defined but not used [-Wunused-function]
     dyn_array_ptr_set (DynPtrArray *da, int x, void *ptr)
     ^~~~~~~~~~~~~~~~~
    
    sgen-dynarray.h:202:1: warning: 'dyn_array_ptr_init' defined but not used [-Wunused-function]
     dyn_array_ptr_init (DynPtrArray *da)
     ^~~~~~~~~~~~~~~~~~
    
    sgen-dynarray.h:194:1: warning: 'dyn_array_int_is_copy' defined but not used [-Wunused-function]
     dyn_array_int_is_copy (DynIntArray *da)
     ^~~~~~~~~~~~~~~~~~~~~
    
    sgen-dynarray.h:188:1: warning: 'dyn_array_int_copy' defined but not used [-Wunused-function]
     dyn_array_int_copy (DynIntArray *dst, DynIntArray *src)
     ^~~~~~~~~~~~~~~~~~
    
    sgen-dynarray.h:182:1: warning: 'dyn_array_int_ensure_independent' defined but not used [-Wunused-function]
     dyn_array_int_ensure_independent (DynIntArray *da)
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    sgen-dynarray.h:168:1: warning: 'dyn_array_int_get' defined but not used [-Wunused-function]
    dyn_array_int_get (DynIntArray *da, int x)
     ^~~~~~~~~~~~~~~~~
    
    sgen-dynarray.h:161:1: warning: 'dyn_array_int_add' defined but not used [-Wunused-function]
     dyn_array_int_add (DynIntArray *da, int x)
     ^~~~~~~~~~~~~~~~~
    
    sgen-dynarray.h:147:1: warning: 'dyn_array_int_size' defined but not used [-Wunused-function]
     dyn_array_int_size (DynIntArray *da)
     ^~~~~~~~~~~~~~~~~~
    sgen-dynarray.h:141:1: warning: 'dyn_array_int_uninit' defined but not used [-Wunused-function]
     dyn_array_int_uninit (DynIntArray *da)
     ^~~~~~~~~~~~~~~~~~~~
    sgen-dynarray.h:135:1: warning: 'dyn_array_int_init' defined but not used [-Wunused-function]
     dyn_array_int_init (DynIntArray *da)
     ^~~~~~~~~~~~~~~~~~

    In file included from sre.c:30:0:
    ../../mono/metadata/reflection-cache.h:43:1: warning: 'free_reflected_entry' defined but not used [-Wunused-function]
     free_reflected_entry (ReflectedEntry *entry)
     ^~~~~~~~~~~~~~~~~~~~

    sre.c:2067:1: warning: 'is_sr_mono_field' defined but not used [-Wunused-function]
     is_sr_mono_field (MonoClass *klass)
     ^~~~~~~~~~~~~~~~

    w32handle.c: In function 'mono_w32handle_wait_multiple':
    w32handle.c:1146:13: warning: 'start' may be used uninitialized in this function [-Wmaybe-uninitialized]
         elapsed = mono_msec_ticks () - start;
         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~

    custom-attrs.c:269:1: warning: 'load_cattr_type_object' defined but not used [-Wunused-function]
     load_cattr_type_object (MonoImage *image, MonoType *t, gboolean header, const char *p, const char *boundp, const char **end, MonoError *error, guint32 *slen)
     ^~~~~~~~~~~~~~~~~~~~~~

    sre-save.c:114:1: warning: 'swap_with_size' defined but not used [-Wunused-function]
     swap_with_size (char *dest, const char* val, int len, int nelem) {


    mono-mmap.c: In function 'mono_file_map_error':
    mono-mmap.c:401:82: warning: format '%llu' expects argument of type 'long long unsigned int', but argument 5 has type 'guint64 {aka long unsigned int}' [-Wformat=]
        *error_message = g_strdup_printf ("%s failed file:%s length:0x%zX offset:0x%lluX error:%s(0x%X)\n",

### CODE REVIEW

File: mono/mini/tramp-arm64-gsharedvt.c [hunk 1]
```diff
@@ -277,7 +277,6 @@ mono_arch_get_gsharedvt_trampoline (MonoTrampInfo **info, gboolean aot)
 	/* Allocate callee register area just below the callee area so it can be accessed from start_gsharedvt_call using negative offsets */
 	/* The + 8 is for alignment */
 	callee_reg_area_offset = 8;
-	callee_stack_area_offset = callee_reg_area_offset + (n_arg_regs * sizeof (target_mgreg_t));
```
Comment (jaykrell) [inline mono/mini/tramp-arm64-gsharedvt.c]: needed?

File: mono/utils/strenc.c [hunk 2]
```diff
@@ -339,12 +339,17 @@ mono_utf8_validate_and_len_with_bounds (const gchar *source, glong max_bytes, gl
 
 		switch (*ptr) {
```
Comment (lateralusX) [inline mono/utils/strenc.c]: Maybe we should make the complete switch statement a little more readable when changing it and make it follow code standard?

File: mono/mini/tramp-arm64-gsharedvt.c (hunk 1 above)
Comment (lateralusX) [reply mono/mini/tramp-arm64-gsharedvt.c]: Doesn't look that way and it looks like it is handled using a bigger offset (both n_arg_regs and n_arg_fregs).

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/utils/strenc.c (hunk 2 above)
Comment (jaykrell) [reply mono/utils/strenc.c]: Agreed.

Comment (jaykrell) [reply mono/utils/strenc.c]: Let's fix this one later -- the warning and the formatting -- I removed this commit.

Comment (jaykrell) [reply mono/utils/strenc.c]: The casts to guchar also aren't needed, and the two functions might be combinable.
The casts were probably that the pointer used to be to char but now is to guchar, so there is no truncation, it is comparing unsigned char to positive int.

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/intrinsics.c [hunk 3]
```diff
@@ -1462,7 +1462,7 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 		 * we avoid folding constants that when computed would raise an error, in
 		 * case the user code was expecting to get that error raised
 		 */
-		if (fsig->param_count == 1 && args [0]->opcode == OP_R8CONST){
+		if (fsig->param_count == 1 && args [0]->opcode == OP_R8CONST) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 4]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			if (c == 'A'){
 				if (strcmp (mname, "Abs") == 0 && fsig->params [0]->type == MONO_TYPE_R8) {
 					opcode = OP_ABS;
-				} else if (strcmp (mname, "Asin") == 0){
+				} else if (strcmp (mname, "Asin") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 5]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					if (fabs (source) <= 1)
 						opcode = OP_ASIN;
-				} else if (strcmp (mname, "Asinh") == 0){
+				} else if (strcmp (mname, "Asinh") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 6]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					opcode = OP_ASINH;
-				} else if (strcmp (mname, "Acos") == 0){
+				} else if (strcmp (mname, "Acos") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 7]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					if (fabs (source) <= 1)
 						opcode = OP_ACOS;
-				} else if (strcmp (mname, "Acosh") == 0){
+				} else if (strcmp (mname, "Acosh") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 8]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					if (source >= 1)
 						opcode = OP_ACOSH;
-				} else if (strcmp (mname, "Atan") == 0){
+				} else if (strcmp (mname, "Atan") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 9]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					opcode = OP_ATAN;
-				} else if (strcmp (mname, "Atanh") == 0){
+				} else if (strcmp (mname, "Atanh") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 10]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					if (fabs (source) < 1)
 						opcode = OP_ATANH;
 				} 
-			} else if (c == 'C'){
+			} else if (c == 'C') {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 11]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 				if (strcmp (mname, "Cos") == 0) {
 					if (!isinf (source))
 						opcode = OP_COS;
-				} else if (strcmp (mname, "Cbrt") == 0){
+				} else if (strcmp (mname, "Cbrt") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 12]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					opcode = OP_CBRT;
-				} else if (strcmp (mname, "Cosh") == 0){
+				} else if (strcmp (mname, "Cosh") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 13]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					opcode = OP_COSH;
 				}
-			} else if (c == 'R'){
+			} else if (c == 'R') {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 14]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 				if (strcmp (mname, "Round") == 0)
 					opcode = OP_ROUND;
-			} else if (c == 'S'){
+			} else if (c == 'S') {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 15]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 				if (strcmp (mname, "Sin") == 0) {
 					if (!isinf (source))
 						opcode = OP_SIN;
 				} else if (strcmp (mname, "Sqrt") == 0) {
 					if (source >= 0)
 						opcode = OP_SQRT;
-				} else if (strcmp (mname, "Sinh") == 0){
+				} else if (strcmp (mname, "Sinh") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 16]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 					opcode = OP_SINH;
 				}
-			} else if (c == 'T'){
+			} else if (c == 'T') {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 17]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 				if (strcmp (mname, "Tan") == 0){
 					if (!isinf (source))
 						opcode = OP_TAN;
-				} else if (strcmp (mname, "Tanh") == 0){
+				} else if (strcmp (mname, "Tanh") == 0) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/mini/intrinsics.c [hunk 18]
```diff
@@ -1471,63 +1471,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 				double *dest = (double *) mono_domain_alloc (cfg->domain, sizeof (double));
-				double result;
+				double result = 0;
 				MONO_INST_NEW (cfg, ins, OP_R8CONST);
 				ins->type = STACK_R8;
 				ins->dreg = mono_alloc_dreg (cfg, (MonoStackType) ins->type);
 				ins->inst_p0 = dest;
 				
-				switch (opcode){
+				switch (opcode) {
```
Comment (lambdageek) [inline mono/mini/intrinsics.c]: line damage

File: mono/metadata/threads.c [hunk 19]
```diff
@@ -6546,7 +6547,7 @@ mono_threads_summarize_execute_internal (MonoContext *ctx, gchar **out, MonoStac
 {
 	static SummarizerGlobalState state;
 
-	int current_idx;
+	int current_idx = 0;
```
Comment (jaykrell) [inline mono/metadata/threads.c]: The logic here is not clear. One use of ‘current_idx’ is guarded by return value of function that sets it, and one is not.

I can remove this commit for now.

@kyte

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/metadata/threads.c (hunk 19 above)
Comment (jaykrell) [reply mono/metadata/threads.c]: Current thread should probably be special case, so current_idx always set.
If it isn't found, abort the dump.
This fixes the case of >128 threads and current thread isn't in the first 128.
Which seems easy enough to produce.

**Pull request merged**